src/components/NetworkDropdown.tsx

```diff
@@ -32,7 +32,14 @@ const Item = ({
 
 const NetworkDropdown = () => {
   const [networkConfig, setNetworkConfig] = useAtom(networkConfigAtom);
-
+  const specifiedNetworkName = new URLSearchParams(window.location.search).get('network');
+  if (specifiedNetworkName !== null && specifiedNetworkName in networkConfigs) {
```

Comment:
consider notifying by `console.error` if someone passing a `network` and it isn't defined in networkConfigs